pkg/vendir/config/lock_config.go

```diff
@@ -114,42 +124,51 @@ func (c LockConfig) FindContents(dirPath, conPath string) (LockDirectoryContents
 		"Expected to find directory '%s' within lock config, but did not", dirPath)
 }
 
-func (c LockConfig) Merge(other LockConfig) error {
+func (c *LockConfig) Merge(other LockConfig) error {
 	for _, dir := range other.Directories {
+		replaced := false
 		for _, con := range dir.Contents {
-			err := c.MergeContents(filepath.Join(dir.Path, con.Path), con)
-			if err != nil {
-				return err
+			replaced = c.ReplaceContents(filepath.Join(dir.Path, con.Path), con)
+			if replaced {
+				continue
+			}
+			replaced = c.AppendContents(dir.Path, con)
+			if replaced {
+				continue
 			}
 		}
+		if !replaced {
+			c.AppendDirectory(dir)
+		}
 	}
 	return nil
 }
 
-func (c LockConfig) MergeContents(path string, replaceCon LockDirectoryContents) error {
-	var matched bool
+func (c *LockConfig) ReplaceContents(path string, replaceCon LockDirectoryContents) bool {
 
 	for i, dir := range c.Directories {
 		for j, con := range dir.Contents {
 			if filepath.Join(dir.Path, con.Path) != path {
 				continue
 			}
-
-			if matched {
-				return fmt.Errorf("Expected to match exactly one directory, but matched multiple")
-			}
-			matched = true
-
-			newCon := replaceCon
-			newCon.Path = con.Path
-
-			dir.Contents[j] = newCon
+			dir.Contents[j] = replaceCon
 			c.Directories[i] = dir
+			return true
 		}
 	}
 
-	if !matched {
-		return fmt.Errorf("Expected to match exactly one directory, but did not match any")
+	return false
+}
+
+func (c LockConfig) AppendContents(path string, appendCon LockDirectoryContents) bool {
```

Comment:
Can we update this function to be `func (c *LockConfig) AppendContents(path string, appendCon LockDirectoryContents) bool` so that it shows that we are changing the lock configuration

Reply:
Sure